src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -70,6 +87,100 @@ public sealed class CoreSession : ICoreSession
         public ICoreServerSession ServerSession => _serverSession;
 
         // public methods
+        /// <inheritdoc />
+        public void AbortTransaction(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            EnsureIsInTransaction(nameof(AbortTransaction));
+
+            try
+            {
+                if (_currentTransaction.StatementId == 0)
+                {
+                    return;
+                }
+
+                try
+                {
+                    var firstAttempt = CreateAbortTransactionOperation();
+                    ExecuteEndTransactionOnPrimary(firstAttempt, cancellationToken);
+                    return;
+                }
+                catch (Exception exception) when (ShouldIgnoreAbortTransactionException(exception))
+                {
+                    return; // ignore exception and return
+                }
+                catch (Exception exception) when (ShouldRetryEndTransactionException(exception))
+                {
+                    // ignore exception and retry
```

Comment:
I'm resolving the ambiguity in the spec the way I expect you to tell me to, that is, ignore ALL exceptions on the second attempt to abort the transaction. As a result, the application will have no way of knowing whether the transaction has been aborted successfully.